cmd/mt-store-cat/main.go

```diff
@@ -0,0 +1,340 @@
+	cassandraTimeout             = flag.Int("cassandra-timeout", 1000, "cassandra timeout in milliseconds")
+	cassandraReadConcurrency     = flag.Int("cassandra-read-concurrency", 20, "max number of concurrent reads to cassandra.")
+	//cassandraWriteConcurrency    = flag.Int("cassandra-write-concurrency", 10, "max number of concurrent writes to cassandra.")
+	cassandraReadQueueSize = flag.Int("cassandra-read-queue-size", 100, "max number of outstanding reads before blocking. value doesn't matter much")
+	//cassandraWriteQueueSize      = flag.Int("cassandra-write-queue-size", 100000, "write queue size per cassandra worker. should be large engough to hold all at least the total number of series expected, divided by how many workers you have")
+	cassandraRetries   = flag.Int("cassandra-retries", 0, "how many times to retry a query before failing it")
+	cqlProtocolVersion = flag.Int("cql-protocol-version", 4, "cql protocol version to use")
+
+	cassandraSSL              = flag.Bool("cassandra-ssl", false, "enable SSL connection to cassandra")
+	cassandraCaPath           = flag.String("cassandra-ca-path", "/etc/raintank/ca.pem", "cassandra CA certificate path when using SSL")
+	cassandraHostVerification = flag.Bool("cassandra-host-verification", true, "host (hostname and server cert) verification when using SSL")
+
+	cassandraAuth     = flag.Bool("cassandra-auth", false, "enable cassandra authentication")
+	cassandraUsername = flag.String("cassandra-username", "cassandra", "username for authentication")
+	cassandraPassword = flag.String("cassandra-password", "cassandra", "password for authentication")
+
+	// our own flags
+	from = flag.String("from", "-24h", "get data from (inclusive)")
+	to   = flag.String("to", "now", "get data until (exclusive)")
+	mdp  = flag.Int("mdp", 0, "max data points to return")
+	fix  = flag.Int("fix", 0, "fix data to this interval like metrictank does quantization")
+)
+
+func printNormal(igens []chunk.IterGen, from, to uint32) {
+	fmt.Println("number of chunks:", len(igens))
+	for i, ig := range igens {
+		fmt.Println("## chunk", i)
+		iter, err := ig.Get()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "chunk %d itergen.Get: %s", i, err)
+			continue
+		}
+		for iter.Next() {
+			ts, val := iter.Values()
+			printRecord(ts, val, ts >= from && ts < to, math.IsNaN(val))
+		}
+	}
+}
+
+func printPointsNormal(points []schema.Point, from, to uint32) {
+	fmt.Println("number of points:", len(points))
+	for _, p := range points {
+		printRecord(p.Ts, p.Val, p.Ts >= from && p.Ts < to, math.IsNaN(p.Val))
+	}
+}
+
+func printRecord(ts uint32, val float64, in, nan bool) {
+	if in {
+		if nan {
+			fmt.Println("> ", time.Unix(int64(ts), 0).Format(tsFormat), "NAN")
+		} else {
+			fmt.Println("> ", time.Unix(int64(ts), 0).Format(tsFormat), val)
+		}
+	} else {
+		if nan {
+			fmt.Println("- ", time.Unix(int64(ts), 0).Format(tsFormat), "NAN")
+		} else {
+			fmt.Println("- ", time.Unix(int64(ts), 0).Format(tsFormat), val)
```

Comment:
personally I'd think it's nicer to just assign something like a `prefix` variable in the `if` / `else` blocks which contains the `-` / `>` and, same for the `val` / `"NAN"`, then remove the (almost) duplicate prints.